seis/ms/blockette.go

```diff
@@ -0,0 +1,148 @@
+package ms
+
+import (
+	"encoding/binary"
+)
+
+const (
+	BlocketteHeaderSize = 4
+	Blockette1000Size   = 4
+	Blockette1001Size   = 4
+)
+
+// BlocketteHeader stores the header of each miniseed blockette.
+type BlocketteHeader struct {
+	BlocketteType uint16
+	NextBlockette uint16 // Byte of next blockette, 0 if last blockette
+}
+
+// DecodeBlocketteHeader returns a BlocketteHeader from a byte slice.
+func DecodeBlocketteHeader(data []byte) BlocketteHeader {
+	var h [BlocketteHeaderSize]byte
+
+	copy(h[:], data)
+
+	return BlocketteHeader{
+		BlocketteType: binary.BigEndian.Uint16(h[0:2]),
+		NextBlockette: binary.BigEndian.Uint16(h[2:4]),
+	}
+}
+
+// EncodeBlocketteHeader converts a BlocketteHeader into a byte slice.
+func EncodeBlocketteHeader(hdr BlocketteHeader) []byte {
+	var b [BlocketteHeaderSize]byte
+
+	binary.BigEndian.PutUint16(b[0:2], hdr.BlocketteType)
+	binary.BigEndian.PutUint16(b[2:4], hdr.NextBlockette)
+
+	h := make([]byte, BlocketteHeaderSize)
+	copy(h[0:BlocketteHeaderSize], b[:])
+
+	return h
+}
+
+// Unmarshal converts a byte slice into the BlocketteHeader
+func (h *BlocketteHeader) Unmarshal(data []byte) error {
+	*h = DecodeBlocketteHeader(data)
```

Comment:
a static func returning the result would be clearer although this approach works fine

Reply:
do you mean,

```golang
func NewBlocketteHeader(data []byte) (*BlocketteHeader, error) {
return DecodeBlocketteHeader(data), nil
}
```

this would work, guess it depends on how it's called. Easily added if needed.